doc/build_apps/logging_cpp.rst

```diff
@@ -200,4 +200,29 @@ Historical state always contains a receipt. Users wishing to implement a receipt
     :language: cpp
     :start-after: SNIPPET_START: get_historical_with_receipt
     :end-before: SNIPPET_END: get_historical_with_receipt
-    :dedent:
\ No newline at end of file
+    :dedent:
+
+User-defined claims in receipts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A user wanting to tie transaction-specific values to a receipt can do so by attaching a claims digest to their transaction:
+
+.. literalinclude:: ../../samples/apps/logging/logging.cpp
+    :language: cpp
+    :start-after: SNIPPET_START: set_claims_digest
+    :end-before: SNIPPET_END: set_claims_digest
+    :dedent:
+
+CCF will then record the digest of the transaction as the combined digest of the write set, plus this claims digest.
+
+Receipts for transactions that have set a claims digest expose a `leaf_components`, rather than an opaque `leaf`,
+which means that a receipt endpoint can choose to reveal the claims and remove their digest from the receipt.
```

Comment:
I think we need to explain why an endpoint would want to do this.

Reply:
I've tried to explain this somewhat generically.